src/dialog/improveAbilityDialog.ts

```diff
@@ -0,0 +1,594 @@
+import { Rune } from "../actors/item-specific/rune";
+import { RqgActor } from "../actors/rqgActor";
+import { Characteristic } from "../data-model/actor-data/characteristics";
+import { ItemTypeEnum } from "../data-model/item-data/itemTypes";
+import { RuneDataProperties } from "../data-model/item-data/runeData";
+import { Ability, IAbility } from "../data-model/shared/ability";
+import { RqgItem } from "../items/rqgItem";
+import { assertItemType, getGame } from "../system/util";
+
+export class ImproveAbilityDialog {
+  //**Shows a dialog for improving a Passion, Rune, or Skill */
+  static async showImproveAbilityDialog(
+    actor: RqgActor,
+    itemId: string,
+    item: Item,
+    speakerName: string
+  ): Promise<void> {
+    const ability = item.data.data as IAbility;
+
+    const adapter: any = {
+      showExperience: ability.hasExperience,
+      showTraining: true,
+      img: item.img,
+      chance: ability.chance || 0,
+      chanceToGain: 100 - Number(ability.chance) || 0,
+      experienceGainFixed: 3,
+      experienceGainRandom: "1d6",
+      trainingGainFixed: 2,
+      trainingGainRandom: "1d6-1",
+    };
+
+    if (item.type === "passion") {
```

Comment:
The typescript typing of foundry likes it better if you check `item.data.type` rather than `item.type` even if they are equivivalent. Also you can check against the Enum instead of just a string. This is in a couple of places, but I only comment on this one.
```suggestion
    if (item.data.type === ItemTypeEnum.Passion) {
```

Reply:
done